packages/components/src/entities/SamplesEditButton.tsx

```diff
@@ -93,6 +99,20 @@ export const SamplesEditButton: FC<OwnProps & SampleGridButtonProps & RequiresMo
         }
     }, [model?.hasSelections, model.id, model.schemaQuery.queryName]);
 
+    const updateSampleHref = useMemo(() => {
+        const updateUrlParam = {
+            target: model?.schemaQuery?.queryName,
+            mode: 'update',
+        };
+
+        let href: any = NEW_SAMPLES_HREF.addParams(updateUrlParam).toHref();
+        if (currentProductId && targetProductId && targetProductId !== currentProductId) {
+            href = createProductUrlFromParts(targetProductId, currentProductId, updateUrlParam, SAMPLES_KEY, 'new');
+        }
+
+        return href;
+    }, [model?.schemaQuery?.queryName]);
```

Comment:
nit. `model.schemaQuery.queryName` is already established above (see line 100) so no need for the defensive null checks.

Also, I believe this dependency array should be `[currentProductId, model.schemaQuery.queryName, targetProductId]`.

Reply:
Looks good.